.github/workflows/publish.yml

```diff
@@ -1,48 +1,36 @@
 name: Publish to pub.dev
 
-# Publish workflow releases a new version, updates changelog
-# and publishes the library to pub.dev
+# Publish workflow checks out specified release tag and publishes
+# the library to pub.dev
 
 on:
-  push:
-    branches:
-      - master
+  workflow_dispatch:
```

Comment:
Well, this would not work, wouldn't it? Since that was the topic of the opened issue in https://github.com/dart-lang/pub-dev/issues/7177 :

From the issue:

> What we would like is to trigger the workflow that publishes the new versions from another workflow by using the workflow_dispatch trigger (**currently only push is allowed**).

Reply:
Managed to trigger new push event from CI, so we can now use `on: push`. Resolving.